plugins/idea/src/main/java/com/oracle/plugin/jtreg/configuration/producers/JTRegClassConfigurationProducer.java

```diff
@@ -59,8 +58,24 @@ protected boolean setupConfigurationFromContext(JTRegConfiguration configuration
         configuration.setWorkingDirectory(JTRegService.getInstance(configuration.getProject()).getWorkDir());
         configuration.setRunClass(psiFile.getVirtualFile().getPath());
         configuration.restoreOriginalModule(originalModule);
-        configuration.setName(psiFile.getName());
+
+        configuration.setQuery(getQuery(element));
+        configuration.setName(nameForElement(element));
+
         initBeforeTaskActions(configuration);
         return true;
     }
+
+    private static String nameForElement(PsiElement element) {
```

Comment:
What are these changes for? These seems unrelated to the binary name problem?

Reply:
These are creating a name for the run configuration. For instance, currently when you create a run configuration, it was always have the same name as the file, but with these changes, when you create a configuration for a method, the name will be e.g. `TestDowncallScope::testDowncall`. This should make it easier to tell apart a run configuration for a whole class from one for an individual method.